llvm/lib/Transforms/Vectorize/VPlanPatternMatch.h

```diff
@@ -290,12 +290,8 @@ struct Recipe_match {
     if ((!matchRecipeAndOpcode<RecipeTys>(R) && ...))
       return false;
 
-    if (R->getNumOperands() != std::tuple_size<Ops_t>::value) {
-      assert(Opcode == Instruction::PHI &&
-             "non-variadic recipe with matched opcode does not have the "
-             "expected number of operands");
+    if (R->getNumOperands() != std::tuple_size<Ops_t>::value)
       return false;
-    }
```

Comment:
This change is due to predicated udivs; not sure if it's worth keeping the assert, permitting udiv:

```
REPLICATE ir<%var2> = udiv ir<%var1>, ir<%i>, ir<%c>
```